ciao-deploy/deploy/utils.go

```
@@ -0,0 +1,317 @@
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "Error running: %v", cmd.Args)
+	}
+	return nil
+}
+
+// SudoRemoveDirectory deletes the directory hiearchy as root
+func SudoRemoveDirectory(ctx context.Context, dest string) error {
+	cmd := SudoCommandContext(ctx, "rm", "-rf", dest)
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "Error running: %v", cmd.Args)
+	}
+	return nil
+}
+
+// SudoRemoveFile deletes the file as root
+func SudoRemoveFile(ctx context.Context, dest string) error {
+	cmd := SudoCommandContext(ctx, "rm", "-f", dest)
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "Error running: %v", cmd.Args)
+	}
+	return nil
+}
+
+// InGoPath returns the desired path relative to $GOPATH
+func InGoPath(path string) string {
+	data, err := exec.Command("go", "env", "GOPATH").Output()
+	gp := ""
+	if err == nil {
+		gp = filepath.Clean(strings.TrimSpace(string(data)))
+	}
+	return filepath.Join(gp, path)
+}
+
+func hostnameWithFallback() string {
+	hs, err := os.Hostname()
+	if err != nil {
+		hs = "localhost"
+	}
+	return hs
+}
+
+// CertName gives file name to use for downloaded certificate
+func CertName(role ssntp.Role) string {
+	return fmt.Sprintf("cert-%s-%s.pem", role.String(), hostnameWithFallback())
+}
+
+// GenerateCert creates a certificate signed by the anchor certificate for a given role
+func GenerateCert(anchorCertPath string, role ssntp.Role) (path string, errOut error) {
+	anchorCertBytes, err := ioutil.ReadFile(anchorCertPath)
+	if err != nil {
+		return "", errors.Wrap(err, "Error reading anchor cert")
+	}
+
+	t, err := certs.CreateCertTemplate(role, "Ciao Deployment", "", []string{}, []string{})
+	if err != nil {
+		return "", errors.Wrap(err, "Error creating certificate template")
+	}
+
+	f, err := ioutil.TempFile("", "cert")
+	if err != nil {
+		return "", errors.Wrap(err, "Error creating temporary certifate file")
+	}
+	defer func() {
+		_ = f.Close()
```

Again, we probably shouldn't ignore this error.